src/commons/editor/Comments.tsx

```diff
@@ -0,0 +1,362 @@
+        if (isCollapsed) {
+          return (
+            <div
+              onClick={setCollapse(comment, false)}
+              style={{
+                backgroundColor: 'rgb(37, 53, 69)',
+                paddingTop: '0.5em',
+                gridColumn: '2'
+              }}
+            >
+              <Divider style={{ borderColor: 'rgba(144, 144, 144, 0.4)' }}></Divider>
+            </div>
+          );
+        }
+        return (
+          <Card className="comment" key={id} style={commentStyles}>
+            <img className="profile-pic" src={profilePic} alt="" style={profilePicStyles}></img>
+            <div className="content" style={contentStyles}>
+              {
+                /* Popover bp-layout="float-right" isn't working */
+                !isEditing ? (
+                  <div style={optionStyles}>
+                    <Popover
+                      content={
+                        <Menu>
+                          <MenuItem text="Edit" onClick={editComment(comment)}></MenuItem>
+                          <MenuItem text="Delete" onClick={deleteComment(comment)}></MenuItem>
+                        </Menu>
+                      }
+                      position={Position.RIGHT_TOP}
+                    >
+                      <Icon icon="more"></Icon>
+                    </Popover>
+                  </div>
+                ) : (
+                  ''
+                )
+              }
+              <div style={optionStyles}>
+                <Icon icon="small-minus" onClick={setCollapse(comment, true)}></Icon>
+              </div>
+              <span className="username" style={usernameStyles}>
+                {username}{' '}
+              </span>
+              <span className="relative-time" style={relativeTimeStyles}>
+                {isUnsubmittedComment(comment) ? 'Preview' : format(new Date(datetime))}
+              </span>
+              <Markdown className="text" content={text || '(Content preview)'} />
+              <div className="error-text" style={errorTextStyles}>
+                {error}
+              </div>
+            </div>
+            {isEditing ? (
+              <div className="reply-container" style={replyContainerStyles}>
+                <textarea
+                  style={enterMessageStyles}
+                  placeholder="Write a message..."
+                  onChange={updatePreviewCommentText(displayComment)}
+                  defaultValue={text}
+                ></textarea>
+                <ButtonGroup>
+                  <Button onClick={cancelWithPrompt(displayComment)}>Cancel</Button>
+                  <Button
+                    intent="success"
+                    onClick={confirmSubmit(displayComment)}
+                    disabled={text.trim().length === 0 || isLoading}
+                  >
+                    Submit
+                  </Button>
+                  {
+                    // Technically not part of button group, by why not?
```

Comment:
If it works, why have more HTML elements just to host a spinner, right?

Reply:
Visual impact. Dynamic movement shows progress/action.

Reply:
(as in, why add more `<div>`s just to contain the spinner) 😆